diff --git a/Makefile.linux b/Makefile.linux
index 6f041b3..ec2f515 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -1,7 +1,7 @@
-uname_P := $(shell sh -c 'uname -p 2>/dev/null || echo not')
+uname_P := $(shell sh -c 'uname -m 2>/dev/null || echo not')
 
 CXX ?= g++
-ISPC ?= ISPC/linux/ispc
+ISPC ?= ispc
 CXXFLAGS ?= -O2 $(ARCH_CXXFLAGS) -fPIC -I.
 ISPC_FLAGS ?= -O2 --arch=$(ISPC_ARCH) --target=$(ISPC_TARGETS) --opt=fast-math --pic
 LDFLAGS ?= -shared -rdynamic
